### PR TITLE
Extended FixedLengthText to include middle elipse and used the FixedLengthText to control comment box project name length.

### PR DESCRIPTION
**Pull request by UlysseFG**

### CODE REVIEW

**Comment by nafiz1001** [comment]
Have you tried following the antd's approach?
https://4x.ant.design/components/typography/#components-typography-demo-ellipsis

File: frontend/src/components/experimentRuns/DatasetArchivedCommentsBox.tsx
Comment (nafiz1001) [inline frontend/src/components/experimentRuns/DatasetArchivedCommentsBox.tsx:1]:
```tsx
  useEffect(() => {
    datasets && setCurrentDataset(datasets[datasetIndex])
}, [datasets])
```
It's missing `datasetIndex' as dependency.

**Comment by UlysseFG** [comment]
> Have you tried following the antd's approach?
Nope. Not sure it existed when i wrote the first one.
https://3x.ant.design/components/typography/#components-typography-demo-ellipsis-middle

**Review by nafiz1001 — APPROVED** [review]
My IDE has won this time.

**Pull request merged**